### PR TITLE
Support for Active Era BS-06 scales

### PR DESCRIPTION
**Pull request by NecroKote**
PR contains rudimentary support for Active Era BS-06 scales.
Scales report weight, feet balance (weight distribution), raw impedance readout from the ADC, and heart rate.
All of those are logged, but only weight is reported in the app. 

Historical measurements are also parsed, but not stored.

The BIA is done inside a native library, which I haven't looked at yet.

### CODE REVIEW

**Comment by oliexdev** [comment]
Thanks for your awesome PR :1st_place_medal:

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    }
+
+    private void parseHistoricalPacket(byte[] pkt) {
+        Instant time = Instant.ofEpochSecond(Converters.fromUnsignedInt24Be(pkt, 3));
+        float weight = (Converters.fromUnsignedInt24Be(pkt, 0x08) & 0x03FFFF) / 1000.0f;
+        float weightLeft = Converters.fromUnsignedInt16Be(pkt, 0x0b) / 100.0f;
+        int hr = pkt[0x0d] & 0xff;
+        int adc = Converters.fromUnsignedInt16Be(pkt, 0x0f);
+        Timber.i("Historical measurement: %.3f kg, Weight Left: %.2f kg, HR: %d, ADC: %d", weight, weightLeft, hr, adc);
+        // TODO: store historical results
```
Comment (oliexdev) [inline android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java:324]: maybe just a simple

```
ScaleMeasurement historyMeasurement = new ScaleMeasurement();
historyMeasurement.setWeight(weight);
addScaleMeasurement(historyMeasurement);
```

would do it?

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+                    // calculate BIA using measure weight and impedance
+                    if (impedance > 0.0) {
+                        final ScaleUser selectedUser = OpenScale.getInstance().getSelectedScaleUser();
+                        int height = (int) Math.ceil(selectedUser.getBodyHeight());
+                        int age = selectedUser.getAge();
+                        int gender = selectedUser.getGender() == Converters.Gender.FEMALE ? 0 : 1;
+
+                        calculateBIA(height, impedance, stableWeightKg, age, gender);
+                        // TODO: report results
```
Comment (oliexdev) [inline android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java:258]: you could set scaleData with the results?

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java (hunk 1 above)
Comment (NecroKote) [reply android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java:324]: The problem is - I'm not the sole user of this scales ) So unless we have a proper "select and claim" mechanic, I would rather not claim everything automatically

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java (hunk 2 above)
Comment (NecroKote) [reply android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java:258]: Here I've used example BIA calculation, just to see how far off the result whould be from what the scales are showing. The results are way off, so this is why I'm not storing the results

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java (hunk 1 above)
Comment (oliexdev) [reply android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java:324]: This should be solved with the settings->Bluetooth->Intelligence user assignment which automatically assign the user which have the nearest weight. You could also enable then ignore out of range measurements to ignore weights that out of the user weights.

File: android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java (hunk 2 above)
Comment (oliexdev) [reply android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothActiveEraBF06.java:258]: ok yeah the vendors calculations are sometimes mysterious. I also implemented some estimates calculations, see https://github.com/oliexdev/openScale/wiki/Body-metric-estimations
You can enable them under settings->measurements->Body fat / Body water / Lean body mass

**Pull request merged**